File: python/같은숫자는싫어.py
def solution(arr):
    answer = []
    
    for i in range(len(arr)):
        if i == 0:
            answer.append(arr[i])
        elif answer[len(answer) - 1] == arr[i]:
            continue
        else :
            answer.append(arr[i])
            
    return answer

def solution(s):
    a = []
    for i in s:
        # 이렇게 하면 a 맨마지막의 list를 리턴 a[-1] 이라고 하면
        # 리스트가 비어있을 때 참조 불가능
        #[i] 이렇게 써주면 리스트로 바꿔줌
        if a[-1:] == [i]: continue
        a.append(i)
    
    return a

File: java/같은_숫자는_싫어.java
import java.util.*;


public class Solution {
    public int[] solution(int []arr) {
        ArrayList<Integer> answer = new ArrayList<>();
        
        for(int i = 0 ; i < arr.length ; i++){
            if(i == 0){ answer.add(arr[i]); continue;}
            
            if(answer.get(answer.size() - 1) != arr[i]) answer.add(arr[i]);
        }

        int toArray[] = answer.stream().mapToInt(i->i).toArray();
        
        return toArray;
    }
}
